src/EscapeAnalysis.jl

```diff
@@ -386,6 +386,25 @@ function escape_builtin!(::typeof(getfield), args::Vector{Any}, pc::Int, state::
     return true
 end
 
+# optimization
+# =======
+
+# similar to IR_FLAG_EFFECT_FREE
+# shifting by 5 and 6 is not used by current implementation (see comment of `ssaflags` in julia.h)
+const IR_FLAG_NO_ESCAPE     = 0x01 << 5
+
+function heap_to_stack_pass!(ir::IRCode, escapes::EscapeState)
+    nstmts = length(ir.stmts)
+    for pc in 1:nstmts
+        # heap-to-stack optimization are carried for heap-allocated objects that are not escaped
```

Comment:
We're only interested in `:new` statement ?

Reply:
I only consider the `:new` stmt, please enlighten me if you feel other cases are also important to this optimization.

Reply:
My point is we can check `isexpr(ir.stmts.inst[pc]], :new` rather than doing these annotations for every statement.

Reply:
oh I see your point!

Reply:
Ummm, simply checking `:new` may not suffice, `ismutabletype` stmts are the ones we are actually looking for I think.

Reply:
Aren't we interested only in `:new` statements for this pass ?

Reply:
I think we are interested in a subset of `:new` statements. I find the below description of `:new`:
>
> Allocates a new struct-like object. First argument is the type. The new pseudo-function is lowered to this, and the type is always inserted by the compiler. This is very much an internal-only feature, and does no checking. Evaluating arbitrary new expressions can easily segfault.

So it does not differentiate mutability IIUC. We are not interested in immutable ones since they are already stack-allocated.